### ISSUE TITLE
src/cfgparse-quic.c: memory corruption suspected by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#2787]
### Tool Name and Version

coverity

### Code Report

```plain
*** CID 1565822:  Memory - corruptions  (BAD_FREE)
/src/cfgparse-quic.c: 228 in bind_parse_quic_cc_algo()
222     
223      out:
224             conf->quic_cc_algo = cc_algo;
225             return 0;
226     
227      fail:
>>>     CID 1565822:  Memory - corruptions  (BAD_FREE)
>>>     "free" frees incorrect pointer "cc_algo".
228             free(cc_algo);
229             return ERR_ALERT | ERR_FATAL;
230     }
231     
232     static int bind_parse_quic_socket(char **args, int cur_arg, struct proxy *px,
233                                       struct bind_conf *conf, char **err)
```

### Output of `haproxy -vv`

```plain
no
```

**Comment by haproxyFred** [comment]
I guess this is in relation with https://github.com/haproxy/haproxy/issues/2786. `<quic_cc_algo_bbr>` could be assigned to `<cc_algo>`, then freed, which is wrong.
So this warning will disappear after @a-denoyelle patch to fix #2786.

**Comment by a-denoyelle** [comment]
Yes it's the same issue which is now already fixed.
